version.sh

```diff
@@ -135,27 +127,40 @@ echo "# Python: $(which python) [$(python --version)]"
 
 seen_mirgecom=0
 
-for i in "${!module_names[@]}"; do
-    url=${module_full_urls[$i]}
-    name=${module_names[$i]}
-    branch=${module_branches[$i]/--branch /}
-    giturl=${url/\#egg=[a-z]*/}
-    [[ ${url} =~ (#egg=[a-z]*) ]] && egg=${BASH_REMATCH[1]} || egg=""
-    [[ -z $url ]] && continue # Ignore non-Git modules
+for name in */; do
+    name=${name/\//}
 
-    if [[ $name == "f2py" ]]; then
-        # Can't install f2py this way
-        continue
-    elif [[ -d $name ]]; then
-        commit=$(cd "$name" && git describe --always)
-    elif [[ $name == "loopy" && -d loo-py ]]; then
-        # FIXME: see above regarding this hack
-        commit=$(cd loo-py && git describe --always)
-    else
+    # Ignore the following dirs
+    [[ $name == "f2py" || $name == "miniforge3" || $name == "config" ]] && continue
+
+    if [[ ! -d "$name" ]]; then
+        ls
         echo "Warning: missing module '$name'. Skipping."
         continue
+    elif [[ ! -d "$name/.git/" ]]; then
+        echo "Warning: not a git module '$name'. Skipping."
+        continue
     fi
 
+    if [[ $name == "loopy" && -d loo-py ]]; then
+        # FIXME: this is a hack that was needed before loo-py was renamed to loopy;
+        # remove this by 12/2021.
+        # https://github.com/illinois-ceesd/emirge/pull/85
+        cd "loo-py" || exit 1
+    else
+        cd "$name" || exit 1
+    fi
+
+    giturl=$(git config --get remote.origin.url)
+    if [[ $giturl == https://* ]]; then
+        giturl="git+$giturl"
+    else
+        giturl="git+ssh://$giturl"
```

Comment:
This isn't true AFAIK. `git+ssh://git@github.com:illinois-ceesd/mirgecom.git` is not a valid git URL.

Reply:
Good catch, fixed.